src/coreclr/tools/Common/Compiler/HardwareIntrinsicHelpers.cs

```diff
@@ -22,7 +22,9 @@ public static bool IsHardwareIntrinsic(MethodDesc method)
             if (owningType.IsIntrinsic && !owningType.HasInstantiation)
             {
                 var owningMdType = (MetadataType)owningType;
-                string ns = owningMdType.ContainingType?.Namespace ?? owningMdType.Namespace;
+                string ns = owningMdType.ContainingType?.ContainingType?.Namespace ??
```

Comment:
Nit: It would be nice to avoid calling the virtual `ContainingType` property multiple times.

Reply:
Fixed